src/optimization/Eliminator.cpp

```diff
@@ -345,6 +346,35 @@ static bool isNoReadBetween(InstructionWalker first, InstructionWalker second, R
 	return true;
 }
 
+void optimizations::translatToMove(const Module& module, Method& method, const Configuration& config)
+{
+	auto it = method.walkAllInstructions();
+	while(!it.isEndOfMethod())
+	{
+		auto const op = it.get<intermediate::Operation>();
+		if (op && (op->op == OP_AND || op->op == OP_OR || op->op == OP_V8MAX || op->op == OP_V8MIN || op->op == OP_MAX || op->op == OP_MIN) && op->getFirstArg() == op->getSecondArg().value()){
+			auto move = new intermediate::MoveOperation(op->getOutput().value(), op->getFirstArg(), op->conditional, op->setFlags);
+			it.erase().emplace(move);
+		}
+
+		it.nextInMethod();
+	}
+}
+
+void optimizations::propagateVar(const Module& module, Method& method, const Configuration& config)
+{
+	auto it = method.walkAllInstructions();
+	while(!it.isEndOfMethod())
+	{
+		auto const op = it.get<intermediate::MoveOperation>();
+		if (op) {
+			it.replace(op->getOutput().value(), op->getSource(), true, true);
```

Comment:
Did you test the effect of this against the regression test-suite?
Though this optimization probably saves a lot of instructions (like all the unnecessary `or x, x, x`), I suspect it will cause a lot of larger kernels to fail due to register-association errors.

Reply:
Sorry, i missed that.
I will fix that, but maybe I need to take a little bit longer time....